DuckDuckGo/AppLifecycle/AppStates/Launching.swift

```diff
@@ -235,19 +235,18 @@ struct Launching: AppState {
         let autoClear = AutoClear(worker: mainViewController)
         self.autoClear = autoClear
         let applicationState = application.applicationState
+        vpnService = VPNService(window: window)
         Task { [vpnService] in
             await autoClear.clearDataIfEnabled(applicationState: .init(with: applicationState))
             await vpnService.installRedditSessionWorkaround()
         }
 
-        unService = UNService(window: window, accountManager: accountManager)
         uiService = UIService(window: window)
 
         // Task handler registration needs to happen before the end of `didFinishLaunching`, otherwise submitting a task can throw an exception.
         // Having both in `didBecomeActive` can sometimes cause the exception when running on a physical device, so registration happens here.
         AppConfigurationFetch.registerBackgroundRefreshTaskHandler()
 
-        UNUserNotificationCenter.current().delegate = unService
```

Comment:
I'm not sure it's the best solution to encapsulate `UserNotificationCenter` inside VPN. Given VPNService conforms to `UNUserNotificationCenterDelegate` there can be `NotificationService` which holds an array of conforming instances, where one of those will be `VPNService`. This way those two are decoupled from each other and we're ready for easier handling of other cases outside of the VPN purpose. It's only a suggestion, since it may increase already quite big scope of the project.

Reply:
I agree, however, I don't want to build a more generic solution for this in this project. If you don't mind I'd leave it in VPN, and leave to a person who wants to start displaying UNNotifications.